recipe/patches/metal_gpu_selection.patch

```diff
@@ -0,0 +1,73 @@
+From f9be8f9492ef7e92a5f169f13d6c4dab83686882 Mon Sep 17 00:00:00 2001
+From: Charles Bousseau <cbousseau@anaconda.com>
+Date: Wed, 17 Apr 2024 16:14:29 -0400
+Subject: [PATCH] metal gpu selection
+
+In macOS, in order for the system to provide a default Metal device object, you must link to the Core Graphics framework. 
+You usually need to do this explicitly if you’re writing apps that don’t use graphics by default, such as command line tools.
+https://developer.apple.com/documentation/metal/1433401-mtlcreatesystemdefaultdevice?language=objc
+Systems with Apple silicon only have one GPU, which removes the need to choose a GPU.
+https://developer.apple.com/documentation/metal/mtldevice/1433409-lowpower#discussion
+---
+ ggml-metal.m | 38 ++++++++++++++++++++++++++++++++++++++
+ 1 file changed, 38 insertions(+)
+
+diff --git a/ggml-metal.m b/ggml-metal.m
+index ae6ddeac..d1383c49 100644
+--- a/ggml-metal.m
++++ b/ggml-metal.m
+@@ -278,6 +278,25 @@ static void ggml_metal_log(enum ggml_log_level level, const char * format, ...){
+ 
+     // Pick and show default Metal device
+     id<MTLDevice> device = MTLCreateSystemDefaultDevice();
++    if (device == nil) {
++        /* 
++        In macOS, in order for the system to provide a default Metal device object, you must link to the Core Graphics framework. 
```

Comment:
Shouldn't we force to link against the core graphics framework? For example, we could do a `find_library(COREGRAPHICS CoreGraphics REQUIRED)` and then append it to the list in https://github.com/ggerganov/llama.cpp/blob/b2687/CMakeLists.txt#L284-L287?

Reply:
I did try something in that line. I'll double check though.

Reply:
I double checked and still get the same behaviour. I don't know why.
I left the existing patch but left a comment at the top of the patch stating this.